front/app/api/map_layers/useUpdateMapLayer.ts

```diff
@@ -9,23 +9,28 @@ import { IMapLayer, IMapLayerUpdateAttributes } from './types';
 
 const updateMapLayer = ({
   id,
-  projectId,
+  mapConfigId,
   ...layer
 }: IMapLayerUpdateAttributes) =>
   fetcher<IMapLayer>({
-    path: `/projects/${projectId}/map_config/layers/${id}`,
+    path: `/map_configs/${mapConfigId}/layers/${id}`,
     action: 'patch',
     body: { layer },
   });
 
-const useUpdateMapLayer = () => {
+const useUpdateMapLayer = (projectId?: string) => {
   const queryClient = useQueryClient();
   return useMutation<IMapLayer, CLErrors, IMapLayerUpdateAttributes>({
     mutationFn: updateMapLayer,
     onSuccess: (_data, variables) => {
       queryClient.invalidateQueries({
-        queryKey: mapConfigKeys.item({ projectId: variables.projectId }),
+        queryKey: mapConfigKeys.item({ mapConfigId: variables.mapConfigId }),
```

Comment:
Since the projectId is optional on the item, you could add that on this line and remove the if condition. Or you move the logic checking for into the spread object
```suggestion
        queryKey: mapConfigKeys.item({ mapConfigId: variables.mapConfigId, projectId }),
```